ansible-scylla-node/tasks/upgrade/main.yml

```diff
@@ -186,120 +196,160 @@
   ansible.builtin.fail:
     msg: "Cross-edition downgrade/upgrade is not implemented yet."
+  any_errors_fatal: True
   when:
     - scylla_upgrade['edition'] != scylla_detected['edition']
     - upgrade_major
     - upgrade_cross_edition
 
+# Determine whether an upgrade actually exists to its target release
+- name: Determine available Enterprise updates for RedHat variants
+  shell: 
+    cmd: yum list available scylla-enterprise | egrep ^scylla-enterprise | awk '{ print $2 }'
+  register: latest_release
+  when: "'scylla-enterprise' in ansible_facts.packages and ansible_os_family == 'RedHat'"
+
+- set_fact: latest_detected_release={{ latest_release.stdout }}
+  when: latest_release.changed
+
+- name: Determine available OSS updates for RedHat variants
+  shell:
+    cmd: yum list available scylla | egrep ^scylla | awk '{ print $2 }'
+  register: latest_release
+  when: "'scylla' in ansible_facts.packages and ansible_os_family == 'RedHat'"
+
+- set_fact: latest_detected_release={{ latest_release.stdout }}
+  when: latest_release.changed
+
+# This check is necessary because when we are on latest under RHEL variants, `yum list available' will return an empty string
+# therefore we pin the fact to its detected_version early on
+- set_fact: latest_detected_release={{ scylla_detected['version'] }}
+  when: ansible_os_family == 'RedHat' and latest_detected_release | length == 0 
+
+- name: Determine available Enterprise updates for Debian variants
+  shell:
+    cmd: apt-cache madison scylla-enterprise | egrep ^scylla-enterprise | awk -F'|' '{ print $2 }' | head -1 | awk '{ print $1 }'
+  register: latest_release
+  when: "'scylla-enterprise' in ansible_facts.packages and ansible_os_family == 'Debian'"
+
+- set_fact: latest_detected_release={{ latest_release.stdout }}
+  when: latest_release.changed
+
+- name: Determine available OSS updates for Debian variants
+  shell:
+    cmd: apt-cache madison scylla | egrep ^scylla | awk -F'|' '{ print $2 }' | head -1 | awk '{ print $1 }'
+  register: latest_release
+  when: "'scylla' in ansible_facts.packages and ansible_os_family == 'Debian'"
+
+- set_fact: latest_detected_release={{ latest_release.stdout }}
+  when: latest_release.changed
+
+- name: Fail when requested version is not within the last release available
+  fail:
+    msg: "Requested upgrade to {{ scylla_upgrade['version'] }} but latest found in repository is {{ latest_detected_release }}"
+  any_errors_fatal: True
+  when: not upgrade_major and scylla_upgrade['version'] is version(latest_detected_release,'>=') and scylla_version != 'latest'
+
+- name: Fail when latest is specified and we are already on latest
+  fail:
+    msg: "'latest' was specified, but currently installed version {{ scylla_detected['version'] }} is the same as last available {{ latest_detected_release }}"
+  any_errors_fatal: True
+  when: not upgrade_major and scylla_detected['version'] is version(latest_detected_release,'>=') and scylla_version == 'latest'
```

Comment:
No, I mean we should detect that there is nothing to upgrade and don't upgrade - simply skip to the next task that follows the upgrade.

The same behavior should be implemented for any "destination" version - not just 'latest'. But this will come naturally the moment you stop special-case 'latest' as I explained above.